.github/workflows/heartbeat-v4.yml

```diff
@@ -0,0 +1,138 @@
+name: heartbeat-v4
+on:
+  workflow_call: # As per documentation, the colon is necessary even though no config is required.
+  workflow_dispatch: # As per documentation, the colon is necessary even though no config is required.
+
+jobs:
```

Comment:
With one exception called out below, this is identical to the body of the old Heartbeat workflow, moved into its own file so it can be invoked via `workflow_dispatch` within the v5 ecosystem (which I imagine we'll want to do until v5 goes GA).